### PR TITLE
[WIP] Add delete role and view, update view listing and add view for an email

### PR DESCRIPTION
**Pull request by vincentmoulene**
**Problem/Motivation**

- Better permissions granularity with a delete permission
- Expose a "view mail" page

**Proposed resolution**

- Add permission to access delete button
- Add view detail page of mail
- Add permission to access new view page
- Write update_hook for new permissions

**Completed tasks**

**Remaining tasks**

- [ ] add new role view permisssion
- [ ] add tests (access to delete button)
- [ ] add tests (access to the view of an email)
- [ ] create update hook to add new permissions to people having the only current one.
- [ ] manualy test the code on an existing project with previous configuration (old views & old permission only)

### CODE REVIEW

**Comment by vints24** [comment]
I think I need to add some tests to close this one.
I will add the differents tests : 
- [ ] access to delete button with differents roles (access when role delete, access denied when another one)
- [ ] access to the view of an email detail (access when role view, access denied when another one)
@WengerK : Do you think it's enough?

**Comment by WengerK** [comment]
Thanks for your greate works on this !

PS: I updated the description to follow the standard drupal template.

File: Templates/backerymails-view.html.twig [hunk 1]
```diff
@@ -0,0 +1,27 @@
+<table>
```
Comment (WengerK) [inline Templates/backerymails-view.html.twig]: the folder name should be lowercase

File: backerymails.module [hunk 2]
```diff
@@ -30,5 +32,17 @@ function backerymails_help($route_name, RouteMatchInterface $route_match) {
...
-  return NULL;
+return NULL;
 }
+
+/**
+ * Implements hook_theme().
+ */
+function backerymails_theme($existing, $type, $theme, $path) {
+  return [
+    'backerymails_view' => [
```
Comment (WengerK) [inline backerymails.module:43]: can we use the same "logic" as node and other entity for template name ?

File: src/Entity/BackerymailsEntity.php [hunk 3]
```diff
@@ -186,6 +185,10 @@ public static function baseFieldDefinitions(EntityTypeInterface $entity_type) {
       ->setLabel(t('Created'))
       ->setDescription(t('The time that the entity was created.'));
 
+    $fields['changed'] = BaseFieldDefinition::create('changed')
+    ->setLabel(t('Changed'))
```
Comment (WengerK) [inline src/Entity/BackerymailsEntity.php]: use `TranslatableMarkup` when possible.

```
->setLabel(new TranslatableMarkup('ID'))
```

File: src/BackerymailsEntityAccessControlHandler.php [hunk 4]
```diff
@@ -24,7 +24,7 @@ protected function checkAccess(EntityInterface $entity, $operation, AccountInter
         return AccessResult::allowedIfHasPermission($account, 'administer backerymails');
```
Comment (WengerK) [inline src/BackerymailsEntityAccessControlHandler.php:24]: Add a Kernel tests for this new permission. You can see an example of this kind of tests here:
https://www.drupal.org/files/issues/2018-12-11/2809291-58.patch

File: backerymails.module (hunk 2 above)
Comment (WengerK) [reply backerymails.module:43]: Example from the node module

```
/**
 * Implements hook_theme().
 */
function node_theme() {
  return [
    'node' => [
      'render element' => 'elements',
    ],
  ];
}
```

**Pull request closed without merging**